core/src/main/java/tc/oc/pgm/spawner/objects/SpawnablePotion.java

```diff
@@ -0,0 +1,42 @@
+package tc.oc.pgm.spawner.objects;
+
+import java.util.List;
+import org.bukkit.Location;
+import org.bukkit.craftbukkit.v1_8_R3.CraftWorld;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.PotionMeta;
+import org.bukkit.potion.Potion;
+import org.bukkit.potion.PotionEffect;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.spawner.Spawnable;
+import tc.oc.pgm.util.nms.NMSHacks;
+
+public class SpawnablePotion implements Spawnable {
+  private final ItemStack potionItem;
+  private final String spawnerId;
+
+  public SpawnablePotion(List<PotionEffect> potion, int damageValue, String spawnerId) {
+    this.spawnerId = spawnerId;
+    // Potion "name" determines potion color
+    ItemStack potionItem = new Potion(damageValue).splash().toItemStack(1);
+    PotionMeta potionMeta = (PotionMeta) potionItem.getItemMeta();
+    for (PotionEffect effect : potion) {
+      // overwrite = false
+      potionMeta.addCustomEffect(effect, false);
+    }
+    potionItem.setItemMeta(potionMeta);
+    this.potionItem = potionItem;
+  }
+
+  @Override
+  public void spawn(Location location, Match match) {
+    NMSHacks.EntityPotion entityPotion = new NMSHacks.EntityPotion(location, potionItem);
+    // TODO set metadata when necessary
+    ((CraftWorld) location.getWorld()).getHandle().addEntity(entityPotion);
```

Comment:
The whole point of NMS hacks is not to do these nms hacks in-line.
if you're needing to import anything from craft-bukkit, that's where it has to be moved to nms hacks (notice you're importing `org.bukkit.craftbukkit.v1_8_R3.CraftWorld;` here)
Create a `NMSHacks.spawnThrownPotion(location, item)` or something like that